### ISSUE TITLE
Heisenwarning `unused_braces` with `tracing`

### ISSUE THREAD

**Issue by Cerber-Ursi** [rust-lang/rust#116347]
### Code

```Rust
#[tracing::instrument]
fn foo() -> u8 { 42 }

#[tracing::instrument]
fn bar() -> u8 { 
    42
}
```

### Current output

```Shell
warning: unnecessary braces around block return value
 --> src/lib.rs:2:16
  |
2 | fn foo() -> u8 { 42 }
  |                ^^  ^^
  |
  = note: `#[warn(unused_braces)]` on by default
help: remove these braces
  |
2 - fn foo() -> u8 { 42 }
2 + fn foo() -> u8 42
  |
```

### Rationale and extra context

I'd expect both `foo` and `bar` to either trigger the warning or not, since they differ only in formatting (in fact, `bar` is just a `rustfmt`-ed `foo`). Looking just at the suggestion, it seems that not warning is better (since this suggestion is not applicable), but this might be a bug in `tracing` and not in `rustc`.

**Comment by workingjubilee** [comment]
The lint does not fire in this case:
```rust
fn foo() -> u8 { 42 }

fn bar() -> u8 { 
    42
}
```

**Comment by p-kraszewski** [comment]
Yes, but it does with `#[tracing::instrument]`, which expands to the _same code_ - and this expanded code somehow behaves differently in `rustc` for single- and multi-line function.

More discussion is here: https://users.rust-lang.org/t/probably-invalid-compiler-hint/100687/1

**Comment by workingjubilee** [comment]
Peculiarly, the lint stops firing with this:
```rust
#[tracing::instrument]
fn foo() -> u8 { dbg!(line!()); 42 }
```

Thus I believe this is a bug in rustc, not in tracing.

**Comment by Cerber-Ursi** [comment]
Possibly-connected problem, in a sense that the code shown after expanding macros is not the same as the code used for compiling (therefore making debug problematic) - https://github.com/dtolnay/cargo-expand/issues/201/

**Comment by shepmaster** [comment]
May be a duplicate of #88104 / #73068 / #70814

**Comment by Cerber-Ursi** [comment]
Seems that this was somehow fixed in a meantime - code from OP does not warn anymore.
